include/zephyr/drivers/spi.h

```diff
@@ -421,6 +421,132 @@ struct spi_buf_set {
 	size_t count;
 };
 
+#if defined(CONFIG_SPI_STATS)
+STATS_SECT_START(spi)
+STATS_SECT_ENTRY32(rx_bytes)
+STATS_SECT_ENTRY32(tx_bytes)
+STATS_SECT_ENTRY32(transfer_error)
+STATS_SECT_END;
+
+STATS_NAME_START(spi)
+STATS_NAME(spi, rx_bytes)
+STATS_NAME(spi, tx_bytes)
+STATS_NAME(spi, transfer_error)
+STATS_NAME_END(spi);
+
+/**
+ * @brief SPI specific device state which allows for SPI device class specific additions
+ */
+struct spi_device_state {
+	struct device_state devstate;
+	struct stats_spi stats;
+};
+
+/**
+ * @brief Get pointer to SPI statistics structure
+ */
+#define Z_SPI_GET_STATS(dev_)				\
+	CONTAINER_OF(dev_->state, struct spi_device_state, devstate)->stats
+
+/**
+ * @brief Increment the rx bytes for a SPI device
+ *
+ * @param dev_ Pointer to the device structure for the driver instance.
+ */
+#define CAN_STATS_RX_BYTES_INC(dev_)			\
```

Comment:
CAN stats?

Reply:
Thanks your review, it's mistake. before i ever moved this commit as draft since realized some internal test is need.